fastavro/writer.py

```diff
@@ -236,6 +238,11 @@ def write_record(fo, datum, schema):
     that they are declared. In other words, a record is encoded as just the
     concatenation of the encodings of its fields.  Field values are encoded per
     their schema."""
+
+    if not validate(datum, schema):
```

Comment:
In the past I didn't want to do this due to speed concerns. You can validate yourself before writing.

I *might* add an optional value `validate` which if set (default to `False`) will validate.